cou/utils/juju_utils.py

```diff
@@ -259,27 +282,18 @@ async def get_status(self) -> FullStatus:
         model = await self._get_model()
         return await model.get_status()
 
-    @retry(no_retry_exceptions=(ActionFailed,))
-    async def _get_action_result(self, action: Action, raise_on_failure: bool) -> Action:
-        """Get results from action.
+    async def list_applications(self) -> dict[str, Application]:
```

Comment:
IMO the naming is confusing. I din't think we should have `list` on it and looking at the first glance it looks like that is all apps and not cou apps
```suggestion
    async def cou_applications(self) -> dict[str, Application]:
```

Reply:
I though about changing it also after pushing, since return value is not list but dict. Renaming it to `cou_applications` do not make much sense, since it will be `COUModel.cou_applications`.

Reply:
What about simply call it `get_apps` or `get_related_apps`?

Reply:
Yeah. I agree that `COUModel.cou_applications` it's not great. I'm ok with `get_apps` or maybe `apps_supported`

Reply:
I will go with `get_supported_apps`, if that's ok with you.

Reply:
Yes. That is ok to me. Thanks